### PR TITLE
Various Google Pay button fixes (3477)

### PR DESCRIPTION
**Pull request by stracker-phil**
### Description

This PR (a) addresses several minor issues with the Google Pay button, and (b) provides a cleaner and more structured JS base class for the payment button.

#### 1. Button disappears when changing shipping method

- On the classic cart, the Google Pay button disappeared instantly when changing the shipping method
- On classic checkout, the Google Pay button eventually disappeared in the _Google Pay Gateway_ when changing the shipping method multiple times

<details>
<summary><strong>📸 Video: Classic Cart</strong></summary>

https://github.com/user-attachments/assets/f510f1f4-5ab0-4c32-ba44-b95d8fe3fbb1

</details>
<details>
<summary><strong>📸 Video: Classic Checkout</strong></summary>

https://github.com/user-attachments/assets/12e9d16a-9d50-4178-b57f-41a38810d985

</details>

#### 2. Block Checkout created duplicate Google Pay instances

- On the block checkout page, two instances of the Google Pay button were created. The reason for this was a duplicate execution context created by some blocks, and resulted in two payment buttons being injected 
  - Already [observed in PCP-2689](https://inpsyde.atlassian.net/browse/PCP-2689?focusedCommentId=300684) (internal link)

<details>
<summary><strong>📸 Screenshot: Payment button logs on block checkout</strong></summary>

</details>

#### 3. Fixed Mini-Cart on product page

- When using Google Pay inside the Mini-Cart while on a product page, the mini-cart now charges for the cart contents, and not the visible product.

<details>
<summary><strong>📸 Screenshot: Mini-Cart on product page</strong></summary>

</details>

#### 4. Button previews in settings

- When changing the button shape in the "Standards Payments" tab, the Google Pay button did not apply the new shape before the page was saved. Now: The Google Pay button updates, like all other PayPal buttons, in real time.

<details>
<summary><strong>📸 Video: Button Preview</strong></summary>

https://github.com/user-attachments/assets/b5906817-e978-4f4c-98f2-6da9e40fce8d

(Note how the Apple Pay preview gets "stuck" once it's changed to "pill" shape once)

</details>

#### 5. Button layout in front-end

- The white Google Pay button correctly shows the dark outline

<details>
<summary><strong>📸 Screenshot: White button in classic checkout</strong></summary>

</details>

### Code structure

- Create a new base class `PaymentButton` 
  - Contains method-independent code, i.e. logic that is also relevant for Apple Pay and other buttons
- Use of JS events to show or hide payment button elements
  - See `PaymentButtonHelpers.js` for [new helpers](https://github.com/woocommerce/woocommerce-paypal-payments/pull/2496/files#diff-f6dc1bc70b9cc91d045ba7829a2a704157610a97d33e041263bdde75a4f482fb).
  - New helpers are used in `CheckoutBootstrap.js` [here](https://github.com/woocommerce/woocommerce-paypal-payments/pull/2496/files#diff-c8cfafba052eed1e40e090694be266dc7090b61185cc87302c0a7240c81d3222R182-R188) and [here](https://github.com/woocommerce/woocommerce-paypal-payments/pull/2496/files#diff-c8cfafba052eed1e40e090694be266dc7090b61185cc87302c0a7240c81d3222R254-R257).
  - Many other small improvements, like the centralized `GooglepayButton.getWrappers()` helper method to pluck CSS IDs from provided config objects, or many new read-only-properties in the `GooglepayButton` instead of public attributes

### CODE REVIEW

**Comment by danieldudzic** [comment]
Great job, really!

One error that I have noticed, is preventing the Google Pay button from loading on the Classic Checkout page:

google_pay

File: modules/ppcp-googlepay/resources/js/GooglepayManager.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ class GooglepayManager {
 				bootstrap.handler
 			);
 
-			const button = new GooglepayButton(
+			const button = GooglepayButton.createButton(
```
Comment (danieldudzic) [inline modules/ppcp-googlepay/resources/js/GooglepayManager.js:23]: Love this change! 💚

**Review by danieldudzic — APPROVED** [review]
This is golden. Awesome work!

**Comment by stracker-phil** [comment]
> ...
> One error that I have noticed, is preventing the Google Pay button from loading on the Classic Checkout page:
> 
> **"No API configuration - missing configure() call?"**

Thanks for reporting

✅ Fixed by this new commit: https://github.com/woocommerce/woocommerce-paypal-payments/pull/2496/commits/966b76102c461e928b8b162ea7f1aaac77ec6582#diff-e0140fc4cf16be408cd39db4c0e6d84cefd19fe55cbe2a2176bd96d9e0fdcc78R289-R291

**Comment by danieldudzic** [comment]
I reviewed and tested this now again, and everything works as expected 🙌

**Pull request merged**